pkg/envoy/ads/jobs.go

```diff
@@ -0,0 +1,47 @@
+package ads
+
+import (
+	"fmt"
+
+	mapset "github.com/deckarep/golang-set"
+	xds_discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/envoy"
+)
+
+// proxyResponseJob is the workerpool job implementation for a Proxy response function
+// It takes the parametres of `server.sendResponse` and allows to queue it as a job on a workerpool
+type proxyResponseJob struct {
+	typeurls  mapset.Set
+	proxy     *envoy.Proxy
+	cfg       *configurator.Configurator
```

Comment:
Making a pointer to an interface type doesn't seem necessary here, especially since the only place this is used it's immediately dereferenced.

Reply:
@eduser25 Here is a very useful doc:  https://github.com/uber-go/guide/blob/master/style.md#pointers-to-interfaces

Reply:
> Making a pointer to an interface type doesn't seem necessary here
This is something I would like to ask for help.

I did it not for necessity, but to avoid creating a new object. Passing the cfg interface by value (I understand) will make a copy of cfg for the callee function's stack, which seems not only undeeded but also expensive in the long run (unless all contents of the struct are prepared for that, ie pointers).
Not sure do we want that for struct-backed interfaces with informers and whatnot?

Reply:
yes... forgot the interfaces are implemented on pointed structs already. Will change.